src/sandstorm/grain.capnp

```diff
@@ -336,6 +336,29 @@ interface SandstormApi(AppObjectId) {
   #   failure.
 }
 
+struct DenormalizedGrainMetadata {
+  # The metadata that we need to present contextual information for shared grains (in particular,
+  # information about the app providing that grain, like icon and title).
+
+  appTitle @0 :Util.LocalizedText;
+  # A copy of the app name for the corresponding UIView for presentation in the grain list.
+
+  union {
+    icon :group {
+      format @1 :Text;
+      # Icon asset format, if present.  One of "png" or "svg"
+
+      assetId @2 :Text;
```

Comment:
Hmm, in order to use `assetId` the app would need to infer the server's static asset host, which seems problematic and possibly even ambiguous depending on the WILDCARD_HOST pattern in use.

I'm not sure what the best alternative is. We don't really have a namespace for assets that apps can manipulate, currently. Complicating matters, we need something that can go both ways (requests and offers).

Arguably the correct representation is actually a capability. Maybe:

    interface StaticAsset {
      getUrl @0 () -> (url :Text);
      getContent @1 () -> (mimeType :Text, data :Data);
    }

(Note: Having `getUrl()` return a string is XSS-prone; we should actually use a representation that can only express protocol schemes that have sane same-origin policy properties -- probably only `http` or `https`.)

And then `SandstormApi` has a new method:

    createStaticAsset (mimeType :Text, content :Data) -> (asset :StaticAsset);

This would allow an app to mint a new asset if needed.

Hopefully this API would turn out useful for other purposes later on.